ethcore/src/signed.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Small utility for signed 256-bit integer.
+
+use std::ops::{Add, AddAssign, Sub, SubAssign};
+use ethereum_types::U256;
+
+/// Sign of a 256-bit integer.
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum Sign {
+	Positive,
+	Zero,
+	Negative
+}
+
+/// Representation of a signed 256-bit integer.
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub struct I256(Sign, U256);
```

Comment:
I would not call it I256 but U256_And_Signed or anything that shows that we are not on 256bit only.